src/components/App/App.jsx

```diff
@@ -0,0 +1,57 @@
+import { useState, useEffect } from "react";
+import "./App.css";
+import { list } from "../../data.js";
+import { TodoList } from "../TodoList/TodoList.jsx";
+import { AddTodoForm } from "../AddTodoForm/AddTodoForm.jsx";
+
+const App = () => {
+	const [todoList, setTodoList] = useState([]);
+
+	const [isLoading, setIsLoading] = useState(true);
+
+	const toLocalStorage = () => {
+		if (isLoading === false) {
+			localStorage.setItem("savedTodoList", JSON.stringify(todoList));
+		}
+	};
+
+	useEffect(() => {
+		new Promise((resolve, reject) => {
+			setTimeout(() => {
+				resolve({
+					data: {
+						todoList: JSON.parse(localStorage.getItem("savedTodoList")) || list,
+					},
+				});
+			}, 500);
+		}).then((result) => {
+			setTodoList(result.data.todoList);
+			setIsLoading(false);
+		});
+	}, []);
+
+	useEffect(toLocalStorage, [todoList]);
```

Comment:
I don't think putting the code here in a function is necessary. Makes your code harder to read, specially since there's nothing else that's invoking `toLocalStorage` here besides this useEffect

Reply:
remade